sklearn/preprocessing/_encoders.py

```diff
@@ -695,6 +697,40 @@ def inverse_transform(self, X):
 
         return X_tr
 
+    def get_feature_names(self, input_features=None):
+        """Return feature names for output features.
+
+        Parameters
+        ----------
+        input_features : list of string, length n_features, optional
+            String names for input features if available. By default,
+            "x0", "x1", ... "xn_features" is used.
+
+        Returns
+        -------
+        output_feature_names : list of string, length n_output_features
+
+        """
+        cats = self.categories_
+        feature_names = []
+        if input_features is None:
+            input_features = ['x%d' % i for i in range(len(cats))]
+        elif(len(input_features) != len(self.categories_)):
+            raise ValueError("input_features should have"
+                             " length equal to number of features")
```

Comment:
error message should probably contain both values, right?
``input_features should have length equal to number of features {}, got {}"``